docker/Dockerfile

```diff
@@ -0,0 +1,244 @@
+ENV EMSCRIPTEN=${EMSDK}/emscripten/sdk
+
+ENV EM_DATA ${EMSDK}/.data
+ENV EM_CONFIG ${EMSDK}/.emscripten
+ENV EM_CACHE ${EM_DATA}/cache
+ENV EM_PORTS ${EM_DATA}/ports
+
+# Fallback in case Emscripten isn't activated
+# Expose Major tools to system PATH, so that emcc, node, asm2wasm etc can be used without activation
+ENV PATH="${EMSDK}:${EMSDK}/emscripten/sdk:${EMSDK}/llvm/clang/bin:${EMSDK}/node/current/bin:${EMSDK}/binaryen/bin:${PATH}"
+
+# Use entrypoint that's coming from emscripten-slim image. It sets all required system paths and variables
+ENTRYPOINT ["/emsdk_portable/entrypoint"]
+
+
+# ------------------------------------------------------------------------------
+# Create a 'standard` 1000:1000 user
+# Thanks to that this image can be executed as non-root user and created files will not require root access level on host machine
+# Please note that this solution even if widely spread (i.e. Node.js uses it) is far from perfect as user 1000:1000 might not exist on
+# host machine, and in this case running any docker image will cause other random problems (mostly due `$HOME` pointing to `/`)
+# This extra user works nicely with entrypoint provided in `/emsdk_portable/entrypoint` as it detects case explained before.
+RUN echo "## Create emscripten user (1000:1000)" \
+    &&  groupadd --gid 1000 emscripten \
+    &&  useradd --uid 1000 --gid emscripten --shell /bin/bash --create-home emscripten \
+    \
+&&  echo "## Done"
+
+# ------------------------------------------------------------------------------
+
+RUN echo "## Update and install packages" \
+# mitigate problem with create symlink to man for base debian image
+    &&  mkdir -p /usr/share/man/man1/ \
+    \
+    &&  apt-get -qq -y update && apt-get -qq install -y --no-install-recommends \
+        libxml2 \
+        ca-certificates \
+        python \
+        python-pip \
+        python3 \
+        python3-pip \
+        wget \
+        curl \
+        zip \
+        unzip \
+        git \
+        ssh-client \
+        build-essential \
+        make \
+        ant \
+        libidn11 \
+        openjdk-11-jre-headless \
+    \
+    # Standard Cleanup on Debian images
+    &&  apt-get -y clean \
+    &&  apt-get -y autoclean \
+    &&  apt-get -y autoremove \
+    &&  rm -rf /var/lib/apt/lists/* \
+    &&  rm -rf /var/cache/debconf/*-old \
+    &&  rm -rf /usr/share/doc/* \
+    &&  rm -rf /usr/share/man/?? \
+    &&  rm -rf /usr/share/man/??_* \
+&&  echo "## Done"
+
+RUN	echo "## Installing CMake" \
+    &&	wget https://cmake.org/files/v3.14/cmake-3.14.3-Linux-x86_64.sh -q \
+    &&	mkdir /opt/cmake \
+    &&	printf "y\nn\n" | sh cmake-3.14.3-Linux-x86_64.sh --prefix=/opt/cmake > /dev/null \
+    &&		rm -fr cmake*.sh /opt/cmake/doc \
+    &&		rm -fr /opt/cmake/bin/cmake-gui \
+    &&		rm -fr /opt/cmake/bin/ccmake \
+    &&		rm -fr /opt/cmake/bin/cpack \
+    &&	ln -s /opt/cmake/bin/cmake /usr/local/bin/cmake \
+    &&	ln -s /opt/cmake/bin/ctest /usr/local/bin/ctest
```

Comment:
Why not just `apt-get install cmake`?

Reply:
To get newer cmake than the one is available for a base distribution: 
```
docker run --rm -it debian:buster bash -c "apt-get update -y && apt-cache policy cmake"
```
```
cmake:
  Installed: (none)
  Candidate: 3.13.4-1
  Version table:
     3.13.4-1 500
        500 http://deb.debian.org/debian buster/main amd64 Packages
```

Reply:
Whats wrong with 3.13.4?  Isn't that a pretty recent version?   Its certainly recent enough to build all the projects related to emscripten.     Maybe this is also left over from when you were building against the older debian version?

Reply:
Two major reasons: 
- back in time base image was Jessy, and there was no decently new cmake available there. 
- It was a contribution: https://github.com/trzecieu/emscripten-docker/pull/43 

For future reasons I thought it will be nice to preserve this privileged to be able to have the most recent version available. 

But no strong feelings about this, I see also value of having a simple apt package being installed.

Reply:
I think now that you are using buster it simpler just to use the packaged version.